moveit_core/online_signal_smoothing/src/butterworth_filter.cpp

```diff
@@ -100,8 +100,11 @@ bool ButterworthFilterPlugin::initialize(rclcpp::Node::SharedPtr node, moveit::c
 {
   node_ = node;
   num_joints_ = num_joints;
-  auto param_listener = std::make_unique<online_signal_smoothing::ParamListener>(node_);
-  auto filter_coeff = param_listener->get_params().butterworth_filter_coeff;
+  double filter_coeff = 1.5;
```

Comment:
i don't think it matters. The `generate_parameters_library` provides a default which is 1.5

Reply:
My thinking was to make it explicit in the code that the value will of `filter_coeff` will **ONLY** be read and set from the parameter file.
Just my opinion on readability.

Reply:
well, the options are:

- don't initialize it at all
- initialize it to 0 or NaN to make it obvious this needs to be set somewhere else. But, these values aren't valid to actually use with the filter.
- initialize it to something valid

I don't think any of the options are great

Reply:
Was wondering how the extra scoping helps here 
```C++
  double filter_coeff = 1.5;
  {
    online_signal_smoothing::ParamListener param_listener(node_);
    filter_coeff = param_listener.get_params().butterworth_filter_coeff;
  }
```
 compared to just

``` c++
    online_signal_smoothing::ParamListener param_listener(node_);
    double filter_coeff = param_listener.get_params().butterworth_filter_coeff;
```
Since in both cases the scope of both `param_listener` and `filter_coeff` is  only `ButterworthFilterPlugin::initialize()`

Sorry about the long conversation.

Reply:
agree, both ways seem fine. If I have time later today, I'll change it to "your way"